crates/typst/src/eval/plugin.rs

```diff
@@ -0,0 +1,59 @@
+use std::ops::Deref;
+use std::sync::Mutex;
+
+use crate::diag::SourceResult;
+use host_wasmer::PluginInstance;
+use typst::diag::At;
+
+#[derive(Debug)]
+pub struct Plugin(pub Mutex<PluginInstance>);
```

Comment:
```suggestion
pub struct Plugin(pub Mutex<PluginInstance>);
```
Ideally the `Mutex` would be private and the plugin exposes it's exact interface through public methods.